types/params.go

```diff
@@ -145,6 +146,12 @@ func (val *ValidatorParams) IsValidPubkeyType(pubkeyType string) bool {
 	return false
 }
 
+func (params *ConsensusParams) Complete() {
+	if params.Synchrony.MessageDelay == 0 && params.Synchrony.Precision == 0 {
```

Comment:
Optional future-proofing:

```go
	if params.Synchrony == (SynchronyParams{}) {
		params.Synchrony = DefaultSynchronyParams()
	}
```

Rationale: Comparing to the zero value ensures we don't forget to check a field if we add one later. YMMV (not clear we would do that anyway)